dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -1789,12 +1793,34 @@ public InputStream searchJSON(Context context, DiscoverQuery discoveryQuery, Str
         solrQuery.setParam(CommonParams.WT, "json");
 
         StringBuilder urlBuilder = new StringBuilder();
-        urlBuilder.append(getSolr().getBaseURL()).append("/select?");
-        urlBuilder.append(solrQuery.toString());
+        //urlBuilder.append(getSolr().getBaseURL()).append("/select?");
+        //urlBuilder.append(solrQuery.toString());
+
+        // New url without any query params appended
+        urlBuilder.append(getSolr().getBaseURL()).append("/select");
+
+        // Post setup
+        NamedList<Object> solrParameters = solrQuery.toNamedList();
+        List<NameValuePair> postParameters = new ArrayList<>();
+        for (Map.Entry<String, Object> solrParameter : solrParameters) {
+            if (solrParameter.getValue() instanceof String[]) {
+                // Multi-valued solr parameter
+                for(String val : (String[])solrParameter.getValue()) {
+                    postParameters.add(new BasicNameValuePair(solrParameter.getKey(), val));
+                }
+
+            }
+            else if(solrParameter.getValue() instanceof String) {postParameters.add(new BasicNameValuePair(solrParameter.getKey(), solrParameter.getValue().toString()));
+            }
+            else {
+                // Is this a problem?
```

Comment:
Just a thought, but should we log a warning here? If we aren't sure if this is a problem, then maybe we should log something to see if it ever gets encountered.